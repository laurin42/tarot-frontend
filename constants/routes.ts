import type { Route } from '@/types/navigation';

/**
 * Typsichere Routenkonstanten für die Navigation
 */
export const APP_ROUTES = {
  TABS: {
    THREECARDS: "/(tabs)/threecards" as Route,
    DAILYCARD: "/(tabs)/dailycard" as Route,
    PROFILE: "/(tabs)/profile" as Route,
  },
  AUTH: "/(auth)" as Route,
  HOME: "/" as Route,
};

/**
 * Konstanten für dynamische Routen
 */
export const DYNAMIC_ROUTES = {
  DEV_TOOLS: "/dev-tools" as const,
};