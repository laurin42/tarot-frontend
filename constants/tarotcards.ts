export interface ITarotCard {
  id: string;
  name: string;
  image: any;
}

export interface ISelectedAndShownCard extends ITarotCard {
  name: string;
  showFront?: boolean;
  isSelected?: boolean;
  currentIndex?: number;
  explanation?: string;
  image: any;
  onNextCard: () => void;
}

export const tarotCards: ITarotCard[] = [
  {
    id: "0",
    name:  "Der Narr",
    image: require("@/assets/images/tarot_cards/The_fool.jpg"),
  },
  {
    id: "1",
    name:  "Der Magier",
    image: require("@/assets/images/tarot_cards/The_magician.jpg"),
  },
  {
    id: "2",
    name: "Die Kaiserin",
    image: require("@/assets/images/tarot_cards/The_empress.jpg"),
  },
  {
    id: "3",
    name: "Die Hohepriesterin",
    image: require("@/assets/images/tarot_cards/High_priestess.jpg"),
  },
  {
    id: "4",
    name: "König der Kelche",
    image: require("@/assets/images/tarot_cards/King_of_cups.jpg"),
  },
  {
    id: "5",
    name: "Ritter der Kelche",
    image: require("@/assets/images/tarot_cards/Knight_of_cups.jpg"),
  },
  {
    id: "6",
    name: "Ritter der Münzen",
    image: require("@/assets/images/tarot_cards/Knight_of_pentacles.jpg"),
  },
  {
    id: "7",
    name: "Page der Kelche",
    image: require("@/assets/images/tarot_cards/Page_of_cups.jpg"),
  },
  {
    id: "8",
    name: "Page der Münzen",
    image: require("@/assets/images/tarot_cards/Page_of_pentacles.jpg"),
  },
  {
    id: "9",
    name: "Königin der Kelche",
    image: require("@/assets/images/tarot_cards/Queen_of_cups.jpg"),
  },
  {
    id: "10",
    name: "Zwei Kelche",
    image: require("@/assets/images/tarot_cards/Two_Cups.jpg"),
  },
  {
    id: "11",
    name: "Drei Kelche",
    image: require("@/assets/images/tarot_cards/Three_Cups.jpg"),
  },
  {
    id: "12",
    name: "Vier Kelche",
    image: require("@/assets/images/tarot_cards/Four_Cups.jpg"),
  },
  {
    id: "13",
    name: "Fünf Kelche",
    image: require("@/assets/images/tarot_cards/Five_Cups.jpg"),
  },
  {
    id: "14",
    name: "Sechs Kelche",
    image: require("@/assets/images/tarot_cards/Six_Cups.jpg"),
  },
  {
    id: "15",
    name: "Sieben Kelche",
    image: require("@/assets/images/tarot_cards/Seven_Cups.jpg"),
  },
  {
    id: "16",
    name: "Acht Kelche",
    image: require("@/assets/images/tarot_cards/Eight_Cups.jpg"),
  },
  {
    id: "17",
    name: "Neun Kelche",
    image: require("@/assets/images/tarot_cards/Nine_Cups.jpg"),
  },
  {
    id: "18",
    name: "Zehn Kelche",
    image: require("@/assets/images/tarot_cards/Ten_Cups.jpg"),
  },
  {
    id: "19",
    name: "Drei Münzen",
    image: require("@/assets/images/tarot_cards/Three_pents.jpg"),
  },
  {
    id: "20",
    name: "Vier Münzen",
    image: require("@/assets/images/tarot_cards/Four_pents.jpg"),
  },
  {
    id: "21",
    name: "Fünf Münzen",
    image: require("@/assets/images/tarot_cards/Five_pents.jpg"),
  },
  {
    id: "22",
    name: "Sechs Münzen",
    image: require("@/assets/images/tarot_cards/Six_pents.jpg"),
  },
  {
    id: "23",
    name: "Sieben Münzen",
    image: require("@/assets/images/tarot_cards/Seven_pents.jpg"),
  },
  {
    id: "24",
    name: "Acht Münzen",
    image: require("@/assets/images/tarot_cards/Eight_pents.jpg"),
  },
  {
    id: "25",
    name: "Neun Münzen",
    image: require("@/assets/images/tarot_cards/Nine_pents.jpg"),
  },
  {
    id: "26",
    name: "Zehn Münzen",
    image: require("@/assets/images/tarot_cards/Ten_pents.jpg"),
  },
  {
    id: "27",
    name: "Ass Münzen",
    image: require("@/assets/images/tarot_cards/Ace_of_Pentacles.jpg"),
  },
];


