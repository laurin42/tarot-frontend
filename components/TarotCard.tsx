import React from "react";
import DynamicTarotCard from "./DynamicTarotCard";
import { ITarotCard } from "../constants/tarotCards";

interface TarotCardProps {
  card: ITarotCard;
  animatedStyle: any;
}

export default function TarotCard({
  card: { name, showFront },
  animatedStyle,
}: TarotCardProps) {
  return (
    <DynamicTarotCard
      cardName={name}
      isShown={showFront ?? true}
      style={animatedStyle}
    />
  );
}
