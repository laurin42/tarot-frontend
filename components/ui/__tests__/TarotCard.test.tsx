import React from "react";
import { render } from "@testing-library/react-native";
import TarotCard from "../../../components/TarotCard";
import { ISelectedAndShownCard } from "@/constants/tarotCards";

const mockCardData: ISelectedAndShownCard = {
  id: "test-card-1",
  name: "Test Card",
  image: "test-image",
  isSelected: false,
  onNextCard: jest.fn(),
};

const mockAnimatedStyle = {
  transform: [{ translateX: 10 }, { translateY: 20 }, { rotate: "5deg" }],
  opacity: 0.8,
  zIndex: 5,
};

describe("TarotCard Component", () => {
  it("renders correctly with card data and matches snapshot", () => {
    const tree = render(
      <TarotCard card={mockCardData} animatedStyle={mockAnimatedStyle} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with card data and animatedStyle and matches snapshot", () => {
    const tree = render(
      <TarotCard card={mockCardData} animatedStyle={mockAnimatedStyle} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when showFront is false and matches snapshot", () => {
    const cardBackData = { ...mockCardData, showFront: false };
    const tree = render(
      <TarotCard card={cardBackData} animatedStyle={mockAnimatedStyle} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
