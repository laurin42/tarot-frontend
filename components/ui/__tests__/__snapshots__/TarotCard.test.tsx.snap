// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TarotCard Component renders correctly when showFront is false and matches snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#000000",
        "borderColor": "black",
        "borderRadius": 16,
        "borderWidth": 0.3,
        "justifyContent": "center",
        "overflow": "hidden",
        "position": "relative",
      },
      {
        "opacity": 0.8,
        "transform": [
          {
            "translateX": 10,
          },
          {
            "translateY": 20,
          },
          {
            "rotate": "5deg",
          },
        ],
        "zIndex": 5,
      },
    ]
  }
>
  <ViewManagerAdapter_ExpoImage
    containerViewRef={"[React.ref]"}
    contentFit="contain"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height="100%"
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    source={
      [
        {
          "testUri": "../../../assets/images/tarot_cards/Card_back_1.png",
        },
      ]
    }
    style={
      {
        "height": "100%",
        "width": "100%",
      }
    }
    transition={
      {
        "duration": 200,
      }
    }
    width="100%"
  />
</View>
`;

exports[`TarotCard Component renders correctly with card data and animatedStyle and matches snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#000000",
        "borderColor": "black",
        "borderRadius": 16,
        "borderWidth": 0.3,
        "justifyContent": "center",
        "overflow": "hidden",
        "position": "relative",
      },
      {
        "opacity": 0.8,
        "transform": [
          {
            "translateX": 10,
          },
          {
            "translateY": 20,
          },
          {
            "rotate": "5deg",
          },
        ],
        "zIndex": 5,
      },
    ]
  }
>
  <ViewManagerAdapter_ExpoImage
    containerViewRef={"[React.ref]"}
    contentFit="contain"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height="100%"
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    source={
      [
        {
          "testUri": "../../../assets/images/tarot_cards/Card_back_1.png",
        },
      ]
    }
    style={
      {
        "height": "100%",
        "width": "100%",
      }
    }
    transition={
      {
        "duration": 200,
      }
    }
    width="100%"
  />
  <View
    style={
      {
        "backgroundColor": "rgba(0, 0, 0, 0.6)",
        "borderRadius": 4,
        "bottom": 4,
        "left": 4,
        "paddingHorizontal": 4,
        "paddingVertical": 2,
        "position": "absolute",
        "right": 4,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 12,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Test Card
    </Text>
  </View>
</View>
`;

exports[`TarotCard Component renders correctly with card data and matches snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#000000",
        "borderColor": "black",
        "borderRadius": 16,
        "borderWidth": 0.3,
        "justifyContent": "center",
        "overflow": "hidden",
        "position": "relative",
      },
      {
        "opacity": 0.8,
        "transform": [
          {
            "translateX": 10,
          },
          {
            "translateY": 20,
          },
          {
            "rotate": "5deg",
          },
        ],
        "zIndex": 5,
      },
    ]
  }
>
  <ViewManagerAdapter_ExpoImage
    containerViewRef={"[React.ref]"}
    contentFit="contain"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height="100%"
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    source={
      [
        {
          "testUri": "../../../assets/images/tarot_cards/Card_back_1.png",
        },
      ]
    }
    style={
      {
        "height": "100%",
        "width": "100%",
      }
    }
    transition={
      {
        "duration": 200,
      }
    }
    width="100%"
  />
  <View
    style={
      {
        "backgroundColor": "rgba(0, 0, 0, 0.6)",
        "borderRadius": 4,
        "bottom": 4,
        "left": 4,
        "paddingHorizontal": 4,
        "paddingVertical": 2,
        "position": "absolute",
        "right": 4,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 12,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Test Card
    </Text>
  </View>
</View>
`;
